src/lib/annotations/doc/DocDrawingThread.js

```diff
@@ -116,16 +114,23 @@ class DocDrawingThread extends DrawingThread {
      * @return {void}
      */
     saveAnnotation(type, text) {
-        super.saveAnnotation(type, text);
+        this.emit('threadcleanup');
```

Comment:
Realized that bindings might be made to bind to 'threaddeleted' so I'll add a custom 'annotationevent' of type 'drawingcommit' to removeAllListeners('annotationevent'). We should be able to clean this up in the future once some logic is pulled out of Annotator.